### ISSUE TITLE
[Feature]: A "pure" python kernel

### ISSUE THREAD

**Issue by silky** [tweag/jupyenv#485]
### What's missing?

The poetry2nix kernel is a bit problematic.

For example, see at least https://github.com/tweag/jupyenv/issues/482, but furthermore, note that sometimes poetry2nix makes certain things quite hard; c.f. tensorflow. It seems hard to get it to work; _but_ `nix-shell -p python310Packages.tensorflow` works fine.

### Possible solution

It'd be great if there was a "pure" python kernel, that just uses the `python3.withPackages ...` style, or something, to just specify the packages in scope.

**Comment by collinarnett** [comment]
Yeah this would be great. The old juypterWith interface was so simple and worked so well. Poetry is a pain to deal with nix leading to infinite recursion errors and hard to debug issues.

**Comment by Princemachiavelli** [comment]
The change to poetry2nix that causes #482 is fairly easy to fix; I just removed the passed poetry parameter for now. But even after using the new poetry2nix version, I get tons of errors like this trying to build the same jupyenv environment (actually it's @collinarnett  stable-diffusion-nix project) which works with the old jupiterWith interface just fine.

```
 Could not find a version that satisfies the requirement setuptools>=41.0.0 (from tensorboard) (from versions: none)
```

**Comment by supermarin** [comment]
Same here, the current state of the project is unusable.

**Comment by GTrunSec** [comment]
See #524
